File: data/EXP01/python/2208/2208.output_1.py
# @lc app=leetcode id=2208 lang=python3
class Solution:
    def halveArray(self, nums: List[int]) -> int:
        total_sum = sum(nums)
        
        nums.sort(reverse=True)
        
        curr_sum = 0
        operations = 0
        for num in nums:
            curr_sum += num
            operations += 1
            if curr_sum >= total_sum / 2:
                break
                
        return operations

File: data/EXP01/golang_no_system_var/2208/2208.output_1.go
func halveArray(nums []int) int {
    sort.Sort(sort.Reverse(sort.IntSlice(nums)))
    
    sum := 0
    for _, num := range nums {
        sum += num
    }
    
    target := sum / 2
    
    count := 0
    currentSum := 0
    for _, num := range nums {
        currentSum += num
        count++
        if currentSum >= target {
            break
        }
    }
    
    return count
}
